### PR TITLE
Make `XLMRoberta` model and config independent from `Roberta`

### PR DESCRIPTION
**Pull request by asofiaoliveira**
# What does this PR do?

Removes the dependencies of `XLMRobertaConfig` and everything inside `modeling_xlm_roberta.py`. This is related to issue #19303.

I only did this for the PyTorch model as there were some models in the issue that had "PyTorch + TF" and this was not one of them. I can add for tensorflow or flax if needed!

## Before submitting
- [x ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

## Who can review?
@sgugger

### CODE REVIEW

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 1]
```diff
@@ -104,13 +1165,98 @@ class XLMRobertaForMaskedLM(RobertaForMaskedLM):
...
+class XLMRobertaForSequenceClassification(XLMRobertaPreTrainedModel):
+    _keys_to_ignore_on_load_missing = [r"position_ids"]
 
-    config_class = XLMRobertaConfig
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.config = config
+
+        self.xlm_roberta = XLMRobertaModel(config, add_pooling_layer=False)
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        self.roberta = XLMRobertaModel(config, add_pooling_layer=False)
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 2]
```diff
@@ -104,13 +1165,98 @@ class XLMRobertaForMaskedLM(RobertaForMaskedLM):
...
+    ) -> Union[Tuple[torch.Tensor], SequenceClassifierOutput]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size,)`, *optional*):
+            Labels for computing the sequence classification/regression loss. Indices should be in `[0, ...,
+            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
+            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.xlm_roberta(
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        outputs = self.roberta(
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 3]
```diff
@@ -120,13 +1266,93 @@ class XLMRobertaForSequenceClassification(RobertaForSequenceClassification):
...
-    """
+# Copied from transformers.models.roberta.modeling_roberta.RobertaForMultipleChoice with Roberta->XLMRoberta, ROBERTA->XLM_ROBERTA
+class XLMRobertaForMultipleChoice(XLMRobertaPreTrainedModel):
+    _keys_to_ignore_on_load_missing = [r"position_ids"]
 
-    config_class = XLMRobertaConfig
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.xlm_roberta = XLMRobertaModel(config)
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        self.roberta = XLMRobertaModel(config)
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 4]
```diff
@@ -120,13 +1266,93 @@ class XLMRobertaForSequenceClassification(RobertaForSequenceClassification):
...
+        flat_position_ids = position_ids.view(-1, position_ids.size(-1)) if position_ids is not None else None
+        flat_token_type_ids = token_type_ids.view(-1, token_type_ids.size(-1)) if token_type_ids is not None else None
+        flat_attention_mask = attention_mask.view(-1, attention_mask.size(-1)) if attention_mask is not None else None
+        flat_inputs_embeds = (
+            inputs_embeds.view(-1, inputs_embeds.size(-2), inputs_embeds.size(-1))
+            if inputs_embeds is not None
+            else None
+        )
+
+        outputs = self.xlm_roberta(
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        outputs = self.roberta(
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 5]
```diff
@@ -136,13 +1362,108 @@ class XLMRobertaForMultipleChoice(RobertaForMultipleChoice):
...
+class XLMRobertaForTokenClassification(XLMRobertaPreTrainedModel):
+    _keys_to_ignore_on_load_unexpected = [r"pooler"]
+    _keys_to_ignore_on_load_missing = [r"position_ids"]
 
-    config_class = XLMRobertaConfig
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.xlm_roberta = XLMRobertaModel(config, add_pooling_layer=False)
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        self.roberta = XLMRobertaModel(config, add_pooling_layer=False)
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 6]
```diff
@@ -136,13 +1362,108 @@ class XLMRobertaForMultipleChoice(RobertaForMultipleChoice):
...
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple[torch.Tensor], TokenClassifierOutput]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Labels for computing the token classification loss. Indices should be in `[0, ..., config.num_labels - 1]`.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.xlm_roberta(
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        outputs = self.roberta(
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 7]
```diff
@@ -152,10 +1473,117 @@ class XLMRobertaForTokenClassification(RobertaForTokenClassification):
...
+# Copied from transformers.models.roberta.modeling_roberta.RobertaForQuestionAnswering with Roberta->XLMRoberta, ROBERTA->XLM_ROBERTA
+class XLMRobertaForQuestionAnswering(XLMRobertaPreTrainedModel):
+    _keys_to_ignore_on_load_unexpected = [r"pooler"]
+    _keys_to_ignore_on_load_missing = [r"position_ids"]
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.xlm_roberta = XLMRobertaModel(config, add_pooling_layer=False)
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        self.roberta = XLMRobertaModel(config, add_pooling_layer=False)
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 8]
```diff
@@ -152,10 +1473,117 @@ class XLMRobertaForTokenClassification(RobertaForTokenClassification):
...
+            Positions are clamped to the length of the sequence (`sequence_length`). Position outside of the sequence
+            are not taken into account for computing the loss.
+        end_positions (`torch.LongTensor` of shape `(batch_size,)`, *optional*):
+            Labels for position (index) of the end of the labelled span for computing the token classification loss.
+            Positions are clamped to the length of the sequence (`sequence_length`). Position outside of the sequence
+            are not taken into account for computing the loss.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.xlm_roberta(
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
        outputs = self.roberta(
```

File: src/transformers/models/xlm_roberta/modeling_xlm_roberta.py [hunk 9]
```diff
@@ -41,6 +63,561 @@
...
+
+# Copied from transformers.models.bert.modeling_roberta.RobertaPreTrainedModel with Roberta->XLMRoberta
+class XLMRobertaPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = XLMRobertaConfig
+    base_model_prefix = "xlm-roberta"
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_xlm_roberta.py]:
```suggestion
    base_model_prefix = "roberta"
```

**Review by sgugger — COMMENTED** [review]
Thanks a lot for working on both of those! The main thing to take care of is that the base model should still be called `roberta` in the models with head.
Other than that, once you're done, make sure to run `make style` on your branch to fix the styling issues.

**Comment by asofiaoliveira** [comment]
Hi @sgugger
I ran `make style` and fixed your recommendations, but now there's an error in the `run_tests_torch`. I believe it's [this](https://app.circleci.com/pipelines/github/huggingface/transformers/48816/workflows/882fa75f-7b2f-45af-861c-1beb9881aeea/jobs/583177?invite=true#step-112-2911): 
```
OSError: Rocketknight1/esm-2-8m is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
If this is a private repository, make sure to pass a token having permission to this repo with `use_auth_token` or log in with `huggingface-cli login` and pass `use_auth_token=True`.
```
I don't really know how to fix that or what caused it.. any tips?

**Review by sgugger — APPROVED** [review]
You can ignore the test, this is a failure we are aware off that will be fixed soon! Looking good at first glance but just want to run a couple more tests to make sure this doesn't break anything. Will do that tomorrow at the latest!

**Review by sgugger — APPROVED** [review]
Tested locally and all is working well, thanks again for your work on this!

**Pull request merged**